test/e2e/common/machinepoolnamelease.go

```diff
@@ -0,0 +1,71 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/hive/pkg/constants"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/client-go/rest"
+	clientcache "k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+)
+
+// WaitForMachinePoolNameLease waits for a lease to appear in the given namespace for the given pool.
+func WaitForMachinePoolNameLease(cfg *rest.Config, namespace, poolName string, timeOut time.Duration) (string, error) {
```

Comment:
This function strikes me as a lot more complicated than it needs to be. Can we accomplish the same thing more easily with `clientwatch.UntilWithSync`?

Reply:
It's done, maybe a little cleaner.